pkg/reconcilermanager/controllers/reposync_controller_manager_test.go

```diff
@@ -93,29 +95,22 @@ func TestRepoSyncReconcilerDeploymentLifecycle(t *testing.T) {
 		t.Fatal("timed out waiting for reconciler deployment to be applied")
 	}
 
-	t.Log("watching for reconciler deployment delete")
-	watchCtx2, watchCancel2 := context.WithTimeout(ctx, 10*time.Second)
-	defer watchCancel2()
-
-	watcher, err = watchObjects(watchCtx2, fakeClient, &appsv1.DeploymentList{})
+	t.Log("verifying the reconciler-manager finalizer is present")
+	rsKey := client.ObjectKeyFromObject(rs)
+	rs = &v1beta1.RepoSync{}
+	err = fakeClient.Get(ctx, rsKey, rs)
 	require.NoError(t, err)
+	require.True(t, controllerutil.ContainsFinalizer(rs, metadata.ReconcilerManagerFinalizer))
 
-	// Delete RootSync
-	rs.ResourceVersion = "" // we don't care what the RV is when deleting
-	err = fakeClient.Delete(ctx, rs)
+	t.Log("deleting sync object and watching for NotFound")
+	err = watchutil.DeleteAndWait(ctx, fakeClient, rs, 10*time.Second)
 	require.NoError(t, err)
 
-	err = watchObjectUntil(ctx, fakeClient.Scheme(), watcher, reconcilerKey, func(event watch.Event) error {
-		t.Logf("reconciler deployment %s", event.Type)
-		if event.Type == watch.Deleted {
-			reconcilerObj = event.Object.(*appsv1.Deployment)
-			// success! deployment was deleted.
-			return nil
-		}
-		// keep watching
-		return errors.Errorf("reconciler deployment %s", event.Type)
-	})
-	require.NoError(t, err)
+	// All managed objects should have been deleted by the reconciler-manager finalizer.
```

Comment:
Should there be an assertion that the managed objects are deleted?

Reply:
That's what Check does.